src/Dapr.Jobs/Models/Responses/JobDetails.cs

```diff
@@ -0,0 +1,89 @@
+﻿// ------------------------------------------------------------------------
+// Copyright 2024 The Dapr Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//     http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ------------------------------------------------------------------------
+
+using System.Text.Json.Serialization;
+using System.Text.RegularExpressions;
+
+namespace Dapr.Jobs.Models.Responses;
+
+/// <summary>
+/// Represents the details of a retrieved job.
+/// </summary>
+public sealed record JobDetails
+{
+    /// <summary>
+    /// Regular expression used to determine if a given schedule is a Cron expression or an interval.
+    /// </summary>
+    private readonly Regex isIntervalRegex = new("h|m|(ms)|s");
+
+    ///<summary>
+    /// A cron-like expression that defines when a job should be triggered.
+    /// </summary>
+    /// <remarks>
+    /// Either this or the <see cref="Interval"/> property should be specified.
+    /// </remarks>
+    public string? CronExpression => Schedule is null || !isIntervalRegex.IsMatch(Schedule) ? Schedule : null;
+
+    /// <summary>
+    /// The interval expression that defines when a job should be triggered.
+    /// </summary>
+    /// <remarks>
+    /// Either this or the <see cref="CronExpression"/> property should be specified.
+    /// </remarks>
+    public TimeSpan? Interval =>
+        Schedule is not null && isIntervalRegex.IsMatch(Schedule) ? Schedule.FromDurationString() : null;
+
+    /// <summary>
+    /// Represents whether the job is scheduled using a Cron expression.
+    /// </summary>
+    public bool IsCronExpression => CronExpression is not null;
```

Comment:
I feel like these should be in a separate `DaprJobSchedule` type that we expose instead of the raw `schedule` property; I'm not a fan of how the schedule-related properties spread out over the job details object.

Reply:
Having thought about it a bit, I agree - I've refactored accordingly. See the latest comment below for details.